DfE.FindInformationAcademiesTrusts/Pages/Trusts/Academies/AcademiesPageModel.cs

```diff
@@ -0,0 +1,38 @@
+﻿using DfE.FindInformationAcademiesTrusts.Data;
+using DfE.FindInformationAcademiesTrusts.Services;
+using DfE.FindInformationAcademiesTrusts.Services.DataSource;
+using DfE.FindInformationAcademiesTrusts.Services.Trust;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DfE.FindInformationAcademiesTrusts.Pages.Trusts.Academies
+{
+    public abstract class AcademiesPageModel(
+        ITrustProvider trustProvider,
+        IDataSourceService dataSourceService,
+        ITrustService trustService,
+        IExportService exportService,
+        ILogger<AcademiesPageModel> logger
+    ) : TrustsAreaModel(trustProvider, dataSourceService, trustService, logger, "Academies in this trust")
+    {
+        protected IExportService ExportService { get; } = exportService;
+
+        public string? TabName { get; init; }
+        public virtual async Task<IActionResult> OnGetExportAsync(string uid)
+        {
+            TrustSummaryServiceModel? trustSummary = await TrustService.GetTrustSummaryAsync(uid);
+            Trust? allAcademiesDetails = await TrustProvider.GetTrustByUidAsync(uid);
+
+            if (trustSummary == null || allAcademiesDetails == null)
+            {
+                return new NotFoundResult();
+            }
+
+            var fileContents = ExportService.ExportAcademiesToSpreadsheetUsingProvider(allAcademiesDetails, trustSummary);
+            string fileName = $"{allAcademiesDetails.Name}-{DateTime.Now.Day}-{DateTime.Now.Month}-{DateTime.Now.Year}.xlsx";
```

Comment:
A couple of questions about the format of the filename (I can see this isn't covered under ACs)
- do we definitely want the date to be d-M-yyyy? Often files are named yyyy-MM-dd for better alphabetical sorting by the file browser
- what behaviour do we want when the Trust Name contains illegal file characters

Reply:
I've jiggled it around and am using yyyy-MM-dd, good call.
I am now sanitising Trust names and have added a test to ensure this is done.